### PR TITLE
PASS1-38: Require PIN entry before displaying seed words

### PR DESCRIPTION
**Pull request by coreylakey**
This branch also will include PASS1-155: Make display of Security Words an optional setting, PASS1-157: Remember PIN number when user selects Shutdown from PIN entry.

### CODE REVIEW

File: ports/stm32/boards/Passport/modules/login_ux.py [hunk 1]
```diff
@@ -286,3 +296,90 @@ async def show(self):
                 utime.sleep(2)
                 system.reset()
                 return
+
+class CheckPinUX(UXStateMachine):
```
Comment (FoundationKen) [inline ports/stm32/boards/Passport/modules/login_ux.py:300]: Can we have this code and the LoginUX code be the same?

File: ports/stm32/boards/Passport/modules/ux.py [hunk 2]
```diff
@@ -1823,7 +1824,7 @@ def draw_security_words_popup(title1, title2, words, y):
                         pin += key
 
         # Decide whether to show the security words
-        if not security_words_confirmed:
+        if not security_words_confirmed and security_words_enabled == 'True':
```
[the remark this answers is not in the document]
Comment (coreylakey) [reply ports/stm32/boards/Passport/modules/ux.py]: Fixed

File: ports/stm32/boards/Passport/modules/constants.py [hunk 3]
```diff
@@ -45,4 +45,7 @@
 UNIT_TYPE_SATS = 1
 
 # Maximum amount of characters in a text entry screen
-MAX_MESSAGE_LEN = 64
\ No newline at end of file
+MAX_MESSAGE_LEN = 64
+
+# Default security words state
+SECURITY_WORDS_ENABLED_DEFAULT = 'True'
```
[the remark this answers is not in the document]
Comment (coreylakey) [reply ports/stm32/boards/Passport/modules/constants.py]: Fixed

File: ports/stm32/boards/Passport/modules/choosers.py [hunk 4]
```diff
@@ -124,4 +124,23 @@ def set_units(idx, text):
 
     return which, ch, set_units
 
+
+def security_words_chooser():
+    from constants import SECURITY_WORDS_ENABLED_DEFAULT
+
+    ch = ['At PIN Entry', 'Never']
+    va = ['True', 'False']
```
[the remark this answers is not in the document]
Comment (coreylakey) [reply ports/stm32/boards/Passport/modules/choosers.py]: Fixed

**Comment by jonathancross** [comment]
@coreylakey is this ready for review?  Looks like the changes requested by @FoundationKen have been made.

**Comment by FoundationKen** [comment]
@jonathancross Thanks for the ping.  This PR includes two separate things that I'd like to PR separately.

1. Enable/disable security words at startup
2. Require the user to enter his or her PIN before viewing the seed words

We're just about to release v1.0.8, so I'm not sure if we can get these in.  Will review with the team.

**Pull request closed without merging**